code/game/gamemodes/changeling/powers/transform.dm

```diff
@@ -11,36 +11,93 @@
 	name = "flesh"
 	flags = NODROP
 
+/obj/item/clothing/glasses/changeling/attack_hand(mob/user)
+	if(loc == user && user.mind && user.mind.changeling)

```

Comment:
i tried that at first, but for some reason the items simply lose their icon and get transferred to the hand instead of deleting